src/tests/Stream/PartitionFilterTest.cpp

```diff
@@ -0,0 +1,165 @@
+  EXPECT_FALSE(UnderTest.shouldStopPartition(KafkaW::PollStatus::Empty));
+  EXPECT_FALSE(UnderTest.hasErrorState());
+}
+
+TEST_F(PartitionFilterTest, ErrorStateRecoveredTimeOut) {
+  EXPECT_FALSE(UnderTest.shouldStopPartition(KafkaW::PollStatus::Error));
+  EXPECT_FALSE(UnderTest.shouldStopPartition(KafkaW::PollStatus::TimedOut));
+  EXPECT_FALSE(UnderTest.hasErrorState());
+}
+
+TEST_F(PartitionFilterTest, ErrorStateRecoveredTimeOutAlt) {
+  EXPECT_FALSE(UnderTest.shouldStopPartition(KafkaW::PollStatus::Error));
+  std::this_thread::sleep_for(40ms);
+  EXPECT_FALSE(UnderTest.shouldStopPartition(KafkaW::PollStatus::TimedOut));
+  EXPECT_FALSE(UnderTest.hasErrorState());
+}
+
+TEST_F(PartitionFilterTest, ErrorStateAndStop) {
+  EXPECT_FALSE(UnderTest.shouldStopPartition(KafkaW::PollStatus::Error));
+  std::this_thread::sleep_for(40ms);
+  EXPECT_TRUE(UnderTest.shouldStopPartition(KafkaW::PollStatus::Error));
+}
+
+TEST_F(PartitionFilterTest, StopTimeNoStopMessage) {
+  UnderTest.setStopTime(std::chrono::system_clock::now() - 5ms);
+  EXPECT_FALSE(UnderTest.shouldStopPartition(KafkaW::PollStatus::Message));
+}
+
+TEST_F(PartitionFilterTest, StopTimeNoStopError) {
+  UnderTest.setStopTime(std::chrono::system_clock::now() - 5ms);
+  EXPECT_FALSE(UnderTest.shouldStopPartition(KafkaW::PollStatus::Error));
+}
+
+TEST_F(PartitionFilterTest, StopTimeNoStopEmpty) {
+  UnderTest.setStopTime(std::chrono::system_clock::now() - 5ms);
+  EXPECT_FALSE(UnderTest.shouldStopPartition(KafkaW::PollStatus::Empty));
+}
+
+TEST_F(PartitionFilterTest, StopTimeNoStopEOP) {
+  UnderTest.setStopTime(std::chrono::system_clock::now() - 5ms);
+  EXPECT_FALSE(
+      UnderTest.shouldStopPartition(KafkaW::PollStatus::EndOfPartition));
+}
+
+TEST_F(PartitionFilterTest, StopTimeNoStopTimeOut) {
+  UnderTest.setStopTime(std::chrono::system_clock::now() - 5ms);
+  EXPECT_FALSE(UnderTest.shouldStopPartition(KafkaW::PollStatus::TimedOut));
+}
+
+TEST_F(PartitionFilterTest, StopTimeNoStopMessageAlt) {
+  UnderTest.setStopTime(std::chrono::system_clock::now() - 15ms);
+  EXPECT_FALSE(UnderTest.shouldStopPartition(KafkaW::PollStatus::Message));
+}
+
+TEST_F(PartitionFilterTest, StopTimeNoStopEmptyAlt) {
+  UnderTest.setStopTime(std::chrono::system_clock::now() - 15ms);
+  EXPECT_FALSE(UnderTest.shouldStopPartition(KafkaW::PollStatus::Empty));
+}
+
+TEST_F(PartitionFilterTest, StopTimeDoStopEOP) {
+  UnderTest.setStopTime(std::chrono::system_clock::now() - 15ms);
+  EXPECT_TRUE(
+      UnderTest.shouldStopPartition(KafkaW::PollStatus::EndOfPartition));
+}
+
+TEST_F(PartitionFilterTest, StopTimeNoStopTimeOutAlt) {
+  UnderTest.setStopTime(std::chrono::system_clock::now() - 15ms);
+  EXPECT_FALSE(UnderTest.shouldStopPartition(KafkaW::PollStatus::TimedOut));
+}
+
+TEST_F(PartitionFilterTest, StopTimeNoStopErrorAlt) {
+  UnderTest.setStopTime(std::chrono::system_clock::now() - 15ms);
+  EXPECT_FALSE(UnderTest.shouldStopPartition(KafkaW::PollStatus::Error));
+}
```

Comment:
Are you missing tests where two errors are passed in consecutively?
I.e. to exercise:
```
case KafkaW::PollStatus::Error:
    if (not HasError) {
      HasError = true;
      ErrorTime = std::chrono::system_clock::now();
    } else if (std::chrono::system_clock::now() > ErrorTime + ErrorTimeOut) {
      return true;
    }
```

Reply:
Nope, I do not. In fact, you commented on that particular test.

Reply:
Ah great, sorry.
Does show the power of good test names though.